### PR TITLE
Fix WriteJdbcPTest.test_supplied_closeable_datasource_is_closed [GH#23859] [HZ-2263]

### PR DESCRIPTION
**Pull request by orcunc**
The test is probably failing because the static member field is shared for two HikariDataSource objects. Instead of static member field use a local list  [HZ-2263]

Fixes : https://hazelcast.atlassian.net/browse/HZ-2263

GH : https://github.com/hazelcast/hazelcast/issues/23859

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

[HZ-2263]: https://hazelcast.atlassian.net/browse/HZ-2263?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by orcunc** [comment]
run-lab-run

**Comment by orcunc** [comment]
run-lab-run

File: hazelcast/src/test/java/com/hazelcast/jet/impl/connector/WriteJdbcPTest.java [hunk 1]
```diff
@@ -176,22 +173,25 @@ public void test() throws SQLException {
     public void test_supplied_closeable_datasource_is_closed() throws SQLException {
         Pipeline p = Pipeline.create();
 
+        CopyOnWriteArrayList<HikariDataSource> hikariDataSourceList = new CopyOnWriteArrayList<>();
```
Comment (arodionov) [inline hazelcast/src/test/java/com/hazelcast/jet/impl/connector/WriteJdbcPTest.java:176]: Why we couldn’t simply use local the local `hikariDataSource` variable here?

File: hazelcast/src/test/java/com/hazelcast/jet/impl/connector/WriteJdbcPTest.java [hunk 2]
```diff
@@ -176,22 +173,25 @@ public void test() throws SQLException {
...
+
         p.readFrom(TestSources.items(IntStream.range(0, PERSON_COUNT).boxed().toArray(Integer[]::new)))
                 .map(item -> entry(item, item.toString()))
                 .writeTo(Sinks.jdbc("INSERT INTO " + tableName + " VALUES(?, ?)",
-                        WriteJdbcPTest::createHikariDataSource,
+                        () -> createHikariDataSource(hikariDataSourceList),
```
Comment (arodionov) [inline hazelcast/src/test/java/com/hazelcast/jet/impl/connector/WriteJdbcPTest.java:181]: And here will be
`() -> hikariDataSource`

File: hazelcast/src/test/java/com/hazelcast/jet/impl/connector/WriteJdbcPTest.java (hunk 1 above)
Comment (orcunc) [reply hazelcast/src/test/java/com/hazelcast/jet/impl/connector/WriteJdbcPTest.java:176]: There are two members in the cluster. Each one is calling the supplier to create a datasource and these are added to the list. At the end of the test there will be two entries in the list

File: hazelcast/src/test/java/com/hazelcast/jet/impl/connector/WriteJdbcPTest.java (hunk 2 above)
Comment (orcunc) [reply hazelcast/src/test/java/com/hazelcast/jet/impl/connector/WriteJdbcPTest.java:181]: The same explanation as above. We need to use a list

**Pull request merged**